sympy/solvers/ode.py

```diff
@@ -310,6 +312,7 @@
     "Liouville",
     "2nd_linear_airy",
     "2nd_linear_bessel",
+    "2nd_hypergeometric",
     "nth_order_reducible",
```

Comment:
```
In [15]: eq = x*(x-1)*f(x).diff(x, 2) + (S(3)/2 -2*x)*f(x).diff(x)

In [16]: classify_ode(eq)
Out[16]: ('2nd_hypergeometric', 'nth_order_reducible', '2nd_power_series_regular')

In [17]: dsolve(eq, hint= 'nth_order_reducible')
Out[17]: 
               ⎛⎧      7/2            5/2           3/2                       
               ⎜⎪     x            5⋅x             x               √x       ac
               ⎜⎪ ─────────── - ──────────── - ──────────── + ─────────── - ──
               ⎜⎪     _______        _______        _______       _______     
               ⎜⎪ 3⋅╲╱ x - 1    12⋅╲╱ x - 1    24⋅╲╱ x - 1    8⋅╲╱ x - 1      
f(x) = C₁ + C₂⋅⎜⎨                                                             
               ⎜⎪        7/2            5/2           3/2                     
               ⎜⎪     ⅈ⋅x          5⋅ⅈ⋅x           ⅈ⋅x             ⅈ⋅√x      ⅈ
               ⎜⎪- ─────────── + ──────────── + ──────────── - ─────────── + ─
               ⎜⎪      _______        _______        _______       _______    
               ⎝⎩  3⋅╲╱ 1 - x    12⋅╲╱ 1 - x    24⋅╲╱ 1 - x    8⋅╲╱ 1 - x     

                      ⎞
osh(√x)               ⎟
───────    for │x│ > 1⎟
  8                   ⎟
                      ⎟
                      ⎟
                      ⎟
⋅asin(√x)             ⎟
─────────   otherwise ⎟
   8                  ⎟
                      ⎠

In [18]: dsolve(eq)
Out[18]: 
           5/2  ┌─  ⎛5/2, -1/2 │  ⎞       ┌─  ⎛0, -3 │  ⎞
       C₁⋅x   ⋅ ├─  ⎜          │ x⎟ + C₂⋅ ├─  ⎜      │ x⎟
               2╵ 1 ⎝   7/2    │  ⎠      2╵ 1 ⎝ -3/2 │  ⎠
f(x) = ──────────────────────────────────────────────────
                          3/4 4 _______                  
                         x   ⋅╲╱ x - 1           
```

Also, `nth_order_reducible` is taking more time.

Reply:
> In [18]: dsolve(eq)

I just tried that example with this PR and got:
```julia
In [1]: eq = x*(x-1)*f(x).diff(x, 2) + (S(3)/2 -2*x)*f(x).diff(x)                                        

In [2]: dsolve(eq)                                                                                       
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-2-42dc4d2e7b25> in <module>
----> 1 dsolve(eq)

~/current/sympy/sympy.git/sympy/solvers/ode.py in dsolve(eq, func, hint, simplify, ics, xi, eta, x0, n, **kwargs)
    677             # The key 'hint' stores the hint needed to be solved for.
    678             hint = hints['hint']
--> 679             return _helper_simplify(eq, hint, hints, simplify, ics=ics)
    680 
    681 def _helper_simplify(eq, hint, match, simplify=True, ics=None, **kwargs):

~/current/sympy/sympy.git/sympy/solvers/ode.py in _helper_simplify(eq, hint, match, simplify, ics, **kwargs)
    702         # attempt to solve for func, and apply any other hint specific
    703         # simplifications
--> 704         sols = solvefunc(eq, func, order, match)
    705         if isinstance(sols, Expr):
    706             rv =  odesimp(eq, sols, func, hint)

~/current/sympy/sympy.git/sympy/solvers/ode.py in ode_2nd_hypergeometric(eq, func, order, match)
   1623             e1 = exp(logcombine(e1, force=True))
   1624 
-> 1625         sol = cancel((e/e1)*x**((-match['k']+1)/2))*sol
   1626         sol = Eq(func, sol)
   1627 

UnboundLocalError: local variable 'e1' referenced before assignment
```

Reply:
Actually some bug is present in the code.

Reply:
now check

Reply:
Okay, that one works now.